app/services/submission/trait_score_parser.rb

```diff
@@ -34,16 +34,20 @@ def call
 
   def map_headers_to_traits
     @upload.log "Mapping file header columns to Trait Descriptors"
-    header = input.readline
+    header = csv.readline
     @trait_names = PlantTrialSubmissionDecorator.decorate(@upload.submission).sorted_trait_names
     @trait_mapping = {}
-    header.split("\t")[1..-1].each_with_index do |column_name, i|
-      next if i >= @trait_names.length
-      column_name = column_name.strip
-      trait_index = @trait_names.find_index(column_name)
-      trait_index = i unless trait_index.present?
-      @trait_mapping[i] = trait_index
-      @upload.log " - Mapped column '#{column_name}' to Trait index #{trait_index} of value #{@trait_names[trait_index]}"
+    if header.blank?
```

Comment:
Maybe - but this piece of submission is soon going to be considerably rebuilt anyway. I'm in the process of discussing with @teatree1212 and @wjurkowski how we are going to extend/change this. So it's not that important right now. I simply wanted to switch from TSV to CSV _before_ we start to add/change upload columns (less work with format change in the future).